// @aeternity/aepp-sdk/es/apis/middleware/models/mappers.js
export const Paths1Q9E32FV3AccountsAccountidAex141TokensGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Q9E32FV3AccountsAccountidAex141TokensGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex141TokenResponse"
            }
          }
        }
      }
    }
  }
};
export const Aex141TokenResponse = {
  type: {
    name: "Composite",
    className: "Aex141TokenResponse",
    modelProperties: {
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      ownerId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "owner_id",
        type: {
          name: "String"
        }
      },
      tokenId: {
        serializedName: "token_id",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PaginatedResponse = {
  type: {
    name: "Composite",
    className: "PaginatedResponse",
    modelProperties: {
      next: {
        serializedName: "next",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      prev: {
        serializedName: "prev",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const ErrorResponse = {
  type: {
    name: "Composite",
    className: "ErrorResponse",
    modelProperties: {
      error: {
        serializedName: "error",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const PathsZ92TkfV3AccountsAccountidAex9BalancesGetResponses200ContentApplicationJsonSchemaAllof1 = {
  type: {
    name: "Composite",
    className: "PathsZ92TkfV3AccountsAccountidAex9BalancesGetResponses200ContentApplicationJsonSchemaAllof1",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex9BalanceResponse"
            }
          }
        }
      }
    }
  }
};
export const Aex9BalanceResponse = {
  type: {
    name: "Composite",
    className: "Aex9BalanceResponse",
    modelProperties: {
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      blockHash: {
        serializedName: "block_hash",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      decimals: {
        serializedName: "decimals",
        required: true,
        type: {
          name: "Number"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      tokenName: {
        serializedName: "token_name",
        required: true,
        type: {
          name: "String"
        }
      },
      tokenSymbol: {
        serializedName: "token_symbol",
        required: true,
        type: {
          name: "String"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      txIndex: {
        serializedName: "tx_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      txType: {
        serializedName: "tx_type",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths1Ymnu9GV3AccountsAccountIdDexSwapsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Ymnu9GV3AccountsAccountIdDexSwapsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "DexSwap"
            }
          }
        }
      }
    }
  }
};
export const DexSwap = {
  type: {
    name: "Composite",
    className: "DexSwap",
    modelProperties: {
      action: {
        serializedName: "action",
        required: true,
        type: {
          name: "String"
        }
      },
      amounts: {
        serializedName: "amounts",
        type: {
          name: "Composite",
          className: "DexSwapAmounts"
        }
      },
      caller: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "caller",
        required: true,
        type: {
          name: "String"
        }
      },
      fromAmount: {
        serializedName: "from_amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      fromContract: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "from_contract",
        required: true,
        type: {
          name: "String"
        }
      },
      fromDecimals: {
        serializedName: "from_decimals",
        required: true,
        type: {
          name: "Number"
        }
      },
      fromToken: {
        serializedName: "from_token",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      toAccount: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "to_account",
        required: true,
        type: {
          name: "String"
        }
      },
      toAmount: {
        serializedName: "to_amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      toContract: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "to_contract",
        required: true,
        type: {
          name: "String"
        }
      },
      toDecimals: {
        serializedName: "to_decimals",
        required: true,
        type: {
          name: "Number"
        }
      },
      toToken: {
        serializedName: "to_token",
        required: true,
        type: {
          name: "String"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const DexSwapAmounts = {
  type: {
    name: "Composite",
    className: "DexSwapAmounts",
    modelProperties: {
      amount0In: {
        serializedName: "amount0_in",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      amount0Out: {
        serializedName: "amount0_out",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      amount1In: {
        serializedName: "amount1_in",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      amount1Out: {
        serializedName: "amount1_out",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      }
    }
  }
};
export const Paths1Ent1R1V3AccountsIdActivitiesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Ent1R1V3AccountsIdActivitiesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Activity"
            }
          }
        }
      }
    }
  }
};
export const Activity = {
  type: {
    name: "Composite",
    className: "Activity",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockTime: {
        serializedName: "block_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      payload: {
        serializedName: "payload",
        type: {
          name: "Composite",
          className: "ActivityPayload"
        }
      },
      type: {
        serializedName: "type",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const ActivityPayload = {
  type: {
    name: "Composite",
    className: "ActivityPayload",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        type: {
          name: "Number"
        }
      },
      encodedTx: {
        constraints: {
          Pattern: new RegExp("^tx_\\w+$")
        },
        serializedName: "encoded_tx",
        type: {
          name: "String"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "hash",
        type: {
          name: "String"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        type: {
          name: "UnixTime"
        }
      },
      signatures: {
        serializedName: "signatures",
        type: {
          name: "Sequence",
          element: {
            constraints: {
              Pattern: new RegExp("^sg_\\w+$")
            },
            type: {
              name: "String"
            }
          }
        }
      },
      tx: {
        serializedName: "tx",
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      amount: {
        serializedName: "amount",
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        type: {
          name: "String"
        }
      },
      logIdx: {
        serializedName: "log_idx",
        type: {
          name: "Number"
        }
      },
      recipientId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "recipient_id",
        type: {
          name: "String"
        }
      },
      senderId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "sender_id",
        type: {
          name: "String"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        type: {
          name: "String"
        }
      },
      recipient: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "recipient",
        type: {
          name: "String"
        }
      },
      sender: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "sender",
        type: {
          name: "String"
        }
      },
      tokenId: {
        serializedName: "token_id",
        type: {
          name: "Number"
        }
      },
      callTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "call_tx_hash",
        type: {
          name: "String"
        }
      },
      contractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "contract_tx_hash",
        type: {
          name: "String"
        }
      },
      function: {
        serializedName: "function",
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        type: {
          name: "Number"
        }
      },
      internalTx: {
        serializedName: "internal_tx",
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      kind: {
        serializedName: "kind",
        type: {
          name: "String"
        }
      },
      refTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "ref_tx_hash",
        type: {
          name: "String"
        }
      }
    }
  }
};
export const PathsC0UvfgV3AccountsIdNamesPointeesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsC0UvfgV3AccountsIdNamesPointeesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Pointee"
            }
          }
        }
      }
    }
  }
};
export const Pointee = {
  type: {
    name: "Composite",
    className: "Pointee",
    modelProperties: {
      active: {
        serializedName: "active",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      blockTime: {
        serializedName: "block_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      key: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "key",
        required: true,
        type: {
          name: "String"
        }
      },
      name: {
        constraints: {
          Pattern: new RegExp("^\\w+\\.chain$")
        },
        serializedName: "name",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const NotFoundResponse = {
  type: {
    name: "Composite",
    className: "NotFoundResponse",
    modelProperties: {
      error: {
        serializedName: "error",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const PathsLv15NfV3TransactionsCountIdGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsLv15NfV3TransactionsCountIdGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      channelCloseMutualTx: {
        serializedName: "channel_close_mutual_tx",
        type: {
          name: "Composite",
          className: "Get200ApplicationJsonProperties"
        }
      },
      channelCloseSoloTx: {
        serializedName: "channel_close_solo_tx",
        type: {
          name: "Composite",
          className: "PathsKcpsuzV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelCloseSoloTx"
        }
      },
      channelCreateTx: {
        serializedName: "channel_create_tx",
        type: {
          name: "Composite",
          className: "Paths1W3C1Z4V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelCreateTx"
        }
      },
      channelDepositTx: {
        serializedName: "channel_deposit_tx",
        type: {
          name: "Composite",
          className: "PathsTvtzmvV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelDepositTx"
        }
      },
      channelForceProgressTx: {
        serializedName: "channel_force_progress_tx",
        type: {
          name: "Composite",
          className: "PathsMxcme6V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelForceProgressTx"
        }
      },
      channelOffchainTx: {
        serializedName: "channel_offchain_tx",
        type: {
          name: "Composite",
          className: "Paths1Cnk4LbV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelOffchainTx"
        }
      },
      channelSetDelegatesTx: {
        serializedName: "channel_set_delegates_tx",
        type: {
          name: "Composite",
          className: "Paths13Ss1Q2V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSetDelegatesTx"
        }
      },
      channelSettleTx: {
        serializedName: "channel_settle_tx",
        type: {
          name: "Composite",
          className: "Paths1Gime4MV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSettleTx"
        }
      },
      channelSlashTx: {
        serializedName: "channel_slash_tx",
        type: {
          name: "Composite",
          className: "PathsVn5L1LV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSlashTx"
        }
      },
      channelSnapshotSoloTx: {
        serializedName: "channel_snapshot_solo_tx",
        type: {
          name: "Composite",
          className: "Paths10T1AqyV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSnapshotSoloTx"
        }
      },
      channelWithdrawTx: {
        serializedName: "channel_withdraw_tx",
        type: {
          name: "Composite",
          className: "PathsTr523PV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelWithdrawTx"
        }
      },
      contractCallTx: {
        serializedName: "contract_call_tx",
        type: {
          name: "Composite",
          className: "PathsChduayV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesContractCallTx"
        }
      },
      contractCreateTx: {
        serializedName: "contract_create_tx",
        type: {
          name: "Composite",
          className: "Paths5Shu25V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesContractCreateTx"
        }
      },
      gaAttachTx: {
        serializedName: "ga_attach_tx",
        type: {
          name: "Composite",
          className: "Paths1RkuxepV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesGaAttachTx"
        }
      },
      gaMetaTx: {
        serializedName: "ga_meta_tx",
        type: {
          name: "Composite",
          className: "PathsQklaaxV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesGaMetaTx"
        }
      },
      nameClaimTx: {
        serializedName: "name_claim_tx",
        type: {
          name: "Composite",
          className: "Paths1Hacjy0V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameClaimTx"
        }
      },
      namePreclaimTx: {
        serializedName: "name_preclaim_tx",
        type: {
          name: "Composite",
          className: "Paths139X1XaV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNamePreclaimTx"
        }
      },
      nameRevokeTx: {
        serializedName: "name_revoke_tx",
        type: {
          name: "Composite",
          className: "Paths1R3Fb8MV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameRevokeTx"
        }
      },
      nameTransferTx: {
        serializedName: "name_transfer_tx",
        type: {
          name: "Composite",
          className: "PathsZdcclfV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameTransferTx"
        }
      },
      nameUpdateTx: {
        serializedName: "name_update_tx",
        type: {
          name: "Composite",
          className: "Paths1VkyqhtV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameUpdateTx"
        }
      },
      oracleExtendTx: {
        serializedName: "oracle_extend_tx",
        type: {
          name: "Composite",
          className: "Paths107D9HzV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleExtendTx"
        }
      },
      oracleQueryTx: {
        serializedName: "oracle_query_tx",
        type: {
          name: "Composite",
          className: "PathsS6Nx2KV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleQueryTx"
        }
      },
      oracleRegisterTx: {
        serializedName: "oracle_register_tx",
        type: {
          name: "Composite",
          className: "Paths184Oz8CV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleRegisterTx"
        }
      },
      oracleResponseTx: {
        serializedName: "oracle_response_tx",
        type: {
          name: "Composite",
          className: "PathsLm5DjtV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleResponseTx"
        }
      },
      payingForTx: {
        serializedName: "paying_for_tx",
        type: {
          name: "Composite",
          className: "Paths16B89KuV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesPayingForTx"
        }
      },
      spendTx: {
        serializedName: "spend_tx",
        type: {
          name: "Composite",
          className: "Paths1Ljyzq7V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesSpendTx"
        }
      },
      total: {
        serializedName: "total",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Get200ApplicationJsonProperties = {
  type: {
    name: "Composite",
    className: "Get200ApplicationJsonProperties",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsKcpsuzV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelCloseSoloTx = {
  type: {
    name: "Composite",
    className: "PathsKcpsuzV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelCloseSoloTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1W3C1Z4V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelCreateTx = {
  type: {
    name: "Composite",
    className: "Paths1W3C1Z4V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelCreateTx",
    modelProperties: {
      initiatorId: {
        serializedName: "initiator_id",
        type: {
          name: "Number"
        }
      },
      responderId: {
        serializedName: "responder_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsTvtzmvV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelDepositTx = {
  type: {
    name: "Composite",
    className: "PathsTvtzmvV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelDepositTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsMxcme6V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelForceProgressTx = {
  type: {
    name: "Composite",
    className: "PathsMxcme6V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelForceProgressTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1Cnk4LbV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelOffchainTx = {
  type: {
    name: "Composite",
    className: "Paths1Cnk4LbV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelOffchainTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths13Ss1Q2V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSetDelegatesTx = {
  type: {
    name: "Composite",
    className: "Paths13Ss1Q2V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSetDelegatesTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1Gime4MV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSettleTx = {
  type: {
    name: "Composite",
    className: "Paths1Gime4MV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSettleTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsVn5L1LV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSlashTx = {
  type: {
    name: "Composite",
    className: "PathsVn5L1LV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSlashTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths10T1AqyV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSnapshotSoloTx = {
  type: {
    name: "Composite",
    className: "Paths10T1AqyV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelSnapshotSoloTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      fromId: {
        serializedName: "from_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsTr523PV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelWithdrawTx = {
  type: {
    name: "Composite",
    className: "PathsTr523PV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesChannelWithdrawTx",
    modelProperties: {
      channelId: {
        serializedName: "channel_id",
        type: {
          name: "Number"
        }
      },
      toId: {
        serializedName: "to_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsChduayV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesContractCallTx = {
  type: {
    name: "Composite",
    className: "PathsChduayV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesContractCallTx",
    modelProperties: {
      callerId: {
        serializedName: "caller_id",
        type: {
          name: "Number"
        }
      },
      contractId: {
        serializedName: "contract_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths5Shu25V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesContractCreateTx = {
  type: {
    name: "Composite",
    className: "Paths5Shu25V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesContractCreateTx",
    modelProperties: {
      ownerId: {
        serializedName: "owner_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1RkuxepV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesGaAttachTx = {
  type: {
    name: "Composite",
    className: "Paths1RkuxepV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesGaAttachTx",
    modelProperties: {
      ownerId: {
        serializedName: "owner_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsQklaaxV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesGaMetaTx = {
  type: {
    name: "Composite",
    className: "PathsQklaaxV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesGaMetaTx",
    modelProperties: {
      gaId: {
        serializedName: "ga_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1Hacjy0V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameClaimTx = {
  type: {
    name: "Composite",
    className: "Paths1Hacjy0V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameClaimTx",
    modelProperties: {
      accountId: {
        serializedName: "account_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths139X1XaV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNamePreclaimTx = {
  type: {
    name: "Composite",
    className: "Paths139X1XaV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNamePreclaimTx",
    modelProperties: {
      accountId: {
        serializedName: "account_id",
        type: {
          name: "Number"
        }
      },
      commitmentId: {
        serializedName: "commitment_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1R3Fb8MV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameRevokeTx = {
  type: {
    name: "Composite",
    className: "Paths1R3Fb8MV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameRevokeTx",
    modelProperties: {
      accountId: {
        serializedName: "account_id",
        type: {
          name: "Number"
        }
      },
      nameId: {
        serializedName: "name_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsZdcclfV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameTransferTx = {
  type: {
    name: "Composite",
    className: "PathsZdcclfV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameTransferTx",
    modelProperties: {
      accountId: {
        serializedName: "account_id",
        type: {
          name: "Number"
        }
      },
      nameId: {
        serializedName: "name_id",
        type: {
          name: "Number"
        }
      },
      recipientId: {
        serializedName: "recipient_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1VkyqhtV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameUpdateTx = {
  type: {
    name: "Composite",
    className: "Paths1VkyqhtV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesNameUpdateTx",
    modelProperties: {
      accountId: {
        serializedName: "account_id",
        type: {
          name: "Number"
        }
      },
      nameId: {
        serializedName: "name_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths107D9HzV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleExtendTx = {
  type: {
    name: "Composite",
    className: "Paths107D9HzV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleExtendTx",
    modelProperties: {
      oracleId: {
        serializedName: "oracle_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsS6Nx2KV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleQueryTx = {
  type: {
    name: "Composite",
    className: "PathsS6Nx2KV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleQueryTx",
    modelProperties: {
      oracleId: {
        serializedName: "oracle_id",
        type: {
          name: "Number"
        }
      },
      senderId: {
        serializedName: "sender_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths184Oz8CV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleRegisterTx = {
  type: {
    name: "Composite",
    className: "Paths184Oz8CV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleRegisterTx",
    modelProperties: {
      accountId: {
        serializedName: "account_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsLm5DjtV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleResponseTx = {
  type: {
    name: "Composite",
    className: "PathsLm5DjtV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesOracleResponseTx",
    modelProperties: {
      oracleId: {
        serializedName: "oracle_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths16B89KuV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesPayingForTx = {
  type: {
    name: "Composite",
    className: "Paths16B89KuV3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesPayingForTx",
    modelProperties: {
      payerId: {
        serializedName: "payer_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1Ljyzq7V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesSpendTx = {
  type: {
    name: "Composite",
    className: "Paths1Ljyzq7V3TransactionsCountIdGetResponses200ContentApplicationJsonSchemaPropertiesSpendTx",
    modelProperties: {
      recipientId: {
        serializedName: "recipient_id",
        type: {
          name: "Number"
        }
      },
      senderId: {
        serializedName: "sender_id",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths8I0YgwV3Aex141GetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths8I0YgwV3Aex141GetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex141Response"
            }
          }
        }
      }
    }
  }
};
export const Aex141Response = {
  type: {
    name: "Composite",
    className: "Aex141Response",
    modelProperties: {
      baseUrl: {
        serializedName: "base_url",
        required: true,
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      contractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "contract_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      creationTime: {
        serializedName: "creation_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      extensions: {
        serializedName: "extensions",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "String"
            }
          }
        }
      },
      invalid: {
        serializedName: "invalid",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      limits: {
        serializedName: "limits",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const PathsWkpcwaV3Aex141ContractidTemplatesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsWkpcwaV3Aex141ContractidTemplatesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex141TemplatesResponse"
            }
          }
        }
      }
    }
  }
};
export const Aex141TemplatesResponse = {
  type: {
    name: "Composite",
    className: "Aex141TemplatesResponse",
    modelProperties: {
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      edition: {
        serializedName: "edition",
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      templateId: {
        serializedName: "template_id",
        required: true,
        type: {
          name: "Number"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const PathsRay4X0V3Aex141ContractidTemplatesTemplateidTokensGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsRay4X0V3Aex141ContractidTemplatesTemplateidTokensGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex141TemplateTokensResponse"
            }
          }
        }
      }
    }
  }
};
export const Aex141TemplateTokensResponse = {
  type: {
    name: "Composite",
    className: "Aex141TemplateTokensResponse",
    modelProperties: {
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      ownerId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "owner_id",
        required: true,
        type: {
          name: "String"
        }
      },
      tokenId: {
        serializedName: "token_id",
        required: true,
        type: {
          name: "Number"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths1TkisghV3Aex141ContractidTokensGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1TkisghV3Aex141ContractidTokensGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex141TokenResponse"
            }
          }
        }
      }
    }
  }
};
export const Paths1Fbvaw8V3Aex141ContractidTokensTokenidGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Fbvaw8V3Aex141ContractidTokensTokenidGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      data: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "data",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths1A8Ah39V3Aex141ContractidTransfersGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1A8Ah39V3Aex141ContractidTransfersGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex141TransferEvent"
            }
          }
        }
      }
    }
  }
};
export const Aex141TransferEvent = {
  type: {
    name: "Composite",
    className: "Aex141TransferEvent",
    modelProperties: {
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      recipient: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "recipient",
        required: true,
        type: {
          name: "String"
        }
      },
      sender: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "sender",
        required: true,
        type: {
          name: "String"
        }
      },
      tokenId: {
        serializedName: "token_id",
        required: true,
        type: {
          name: "Number"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths1Uqqby0V3Aex9GetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Uqqby0V3Aex9GetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex9Response"
            }
          }
        }
      }
    }
  }
};
export const Aex9Response = {
  type: {
    name: "Composite",
    className: "Aex9Response",
    modelProperties: {
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      contractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "contract_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      decimals: {
        serializedName: "decimals",
        required: true,
        type: {
          name: "Number"
        }
      },
      eventSupply: {
        serializedName: "event_supply",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      extensions: {
        serializedName: "extensions",
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "String"
            }
          }
        }
      },
      holders: {
        serializedName: "holders",
        required: true,
        type: {
          name: "Number"
        }
      },
      initialSupply: {
        serializedName: "initial_supply",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      invalid: {
        serializedName: "invalid",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      name: {
        serializedName: "name",
        required: true,
        type: {
          name: "String"
        }
      },
      symbol: {
        serializedName: "symbol",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths19IxhsmV3Aex9CountGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths19IxhsmV3Aex9CountGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsEeiffwV3Aex9ContractidBalancesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsEeiffwV3Aex9ContractidBalancesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Aex9ContractBalanceResponse"
            }
          }
        }
      }
    }
  }
};
export const Aex9ContractBalanceResponse = {
  type: {
    name: "Composite",
    className: "Aex9ContractBalanceResponse",
    modelProperties: {
      accountId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "account_id",
        required: true,
        type: {
          name: "String"
        }
      },
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      blockHash: {
        serializedName: "block_hash",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractId: {
        serializedName: "contract_id",
        required: true,
        type: {
          name: "Number"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      lastLogIdx: {
        serializedName: "last_log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      lastTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "last_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const PathsKr825V3Aex9ContractidBalancesAccountidGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsKr825V3Aex9ContractidBalancesAccountidGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      account: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "account",
        required: true,
        type: {
          name: "String"
        }
      },
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      contract: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths108B3VtV3Aex9ContractidBalancesAccountidHistoryGetResponses200ContentApplicationJsonSchemaAllof1 = {
  type: {
    name: "Composite",
    className: "Paths108B3VtV3Aex9ContractidBalancesAccountidHistoryGetResponses200ContentApplicationJsonSchemaAllof1",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Get200ApplicationJsonAllOfPropertiesItemsItem"
            }
          }
        }
      }
    }
  }
};
export const Get200ApplicationJsonAllOfPropertiesItemsItem = {
  type: {
    name: "Composite",
    className: "Get200ApplicationJsonAllOfPropertiesItemsItem",
    modelProperties: {
      account: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "account",
        required: true,
        type: {
          name: "String"
        }
      },
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      contract: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsQmewnaV3ChannelsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsQmewnaV3ChannelsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Channel"
            }
          }
        }
      }
    }
  }
};
export const Channel = {
  type: {
    name: "Composite",
    className: "Channel",
    modelProperties: {
      active: {
        serializedName: "active",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      channel: {
        constraints: {
          Pattern: new RegExp("^ch_\\w{38,50}$")
        },
        serializedName: "channel",
        required: true,
        type: {
          name: "String"
        }
      },
      channelReserve: {
        serializedName: "channel_reserve",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      delegateIds: {
        serializedName: "delegate_ids",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      initiator: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "initiator",
        required: true,
        type: {
          name: "String"
        }
      },
      initiatorAmount: {
        serializedName: "initiator_amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      lastUpdatedHeight: {
        serializedName: "last_updated_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      lastUpdatedTime: {
        serializedName: "last_updated_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      lastUpdatedTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "last_updated_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      lastUpdatedTxType: {
        serializedName: "last_updated_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      lockPeriod: {
        serializedName: "lock_period",
        required: true,
        type: {
          name: "Number"
        }
      },
      lockedUntil: {
        serializedName: "locked_until",
        required: true,
        type: {
          name: "Number"
        }
      },
      responder: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "responder",
        required: true,
        type: {
          name: "String"
        }
      },
      responderAmount: {
        serializedName: "responder_amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      round: {
        serializedName: "round",
        required: true,
        type: {
          name: "Number"
        }
      },
      soloRound: {
        serializedName: "solo_round",
        required: true,
        type: {
          name: "Number"
        }
      },
      stateHash: {
        constraints: {
          Pattern: new RegExp("^st_\\w+$")
        },
        serializedName: "state_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      updatesCount: {
        serializedName: "updates_count",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths18L84JcV3ContractsCallsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths18L84JcV3ContractsCallsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "ContractCall"
            }
          }
        }
      }
    }
  }
};
export const ContractCall = {
  type: {
    name: "Composite",
    className: "ContractCall",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      callTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "call_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      contractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "contract_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      function: {
        serializedName: "function",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      internalTx: {
        serializedName: "internal_tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      localIdx: {
        serializedName: "local_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths10Kk1UxV3ContractsLogsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths10Kk1UxV3ContractsLogsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "ContractLog"
            }
          }
        }
      }
    }
  }
};
export const ContractLog = {
  type: {
    name: "Composite",
    className: "ContractLog",
    modelProperties: {
      args: {
        serializedName: "args",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "String"
            }
          }
        }
      },
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockTime: {
        serializedName: "block_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      callTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "call_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      contractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "contract_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "String"
        }
      },
      eventHash: {
        serializedName: "event_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      eventName: {
        serializedName: "event_name",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      extCallerContractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "ext_caller_contract_id",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      extCallerContractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "ext_caller_contract_tx_hash",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      parentContractId: {
        serializedName: "parent_contract_id",
        required: true,
        nullable: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Contract = {
  type: {
    name: "Composite",
    className: "Contract",
    modelProperties: {
      aexnType: {
        serializedName: "aexn_type",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      contract: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract",
        required: true,
        type: {
          name: "String"
        }
      },
      createTx: {
        serializedName: "create_tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths9Yfxl2V3DexSwapsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths9Yfxl2V3DexSwapsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "DexSwap"
            }
          }
        }
      }
    }
  }
};
export const Paths6Vze8ZV3DexContractIdSwapsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths6Vze8ZV3DexContractIdSwapsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "DexSwap"
            }
          }
        }
      }
    }
  }
};
export const PathsEue6HzV3KeyBlocksGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsEue6HzV3KeyBlocksGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "KeyBlock"
            }
          }
        }
      }
    }
  }
};
export const KeyBlock = {
  type: {
    name: "Composite",
    className: "KeyBlock",
    modelProperties: {
      beneficiary: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "beneficiary",
        required: true,
        type: {
          name: "String"
        }
      },
      beneficiaryReward: {
        serializedName: "beneficiary_reward",
        type: {
          name: "Number"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^kh_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      info: {
        constraints: {
          Pattern: new RegExp("^cb_\\w+$")
        },
        serializedName: "info",
        required: true,
        type: {
          name: "String"
        }
      },
      microBlocksCount: {
        serializedName: "micro_blocks_count",
        required: true,
        type: {
          name: "Number"
        }
      },
      miner: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "miner",
        required: true,
        type: {
          name: "String"
        }
      },
      nonce: {
        serializedName: "nonce",
        type: {
          name: "Number"
        }
      },
      pow: {
        serializedName: "pow",
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Number"
            }
          }
        }
      },
      prevHash: {
        serializedName: "prev_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      prevKeyHash: {
        constraints: {
          Pattern: new RegExp("^kh_\\w{38,50}$")
        },
        serializedName: "prev_key_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      stateHash: {
        constraints: {
          Pattern: new RegExp("^bs_\\w{38,50}$")
        },
        serializedName: "state_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      target: {
        serializedName: "target",
        required: true,
        type: {
          name: "Number"
        }
      },
      time: {
        serializedName: "time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      transactionsCount: {
        serializedName: "transactions_count",
        required: true,
        type: {
          name: "Number"
        }
      },
      version: {
        serializedName: "version",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsNn60D7V3KeyBlocksHashOrKbiMicroBlocksGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsNn60D7V3KeyBlocksHashOrKbiMicroBlocksGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "MicroBlock"
            }
          }
        }
      }
    }
  }
};
export const MicroBlock = {
  type: {
    name: "Composite",
    className: "MicroBlock",
    modelProperties: {
      hash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      microBlockIndex: {
        serializedName: "micro_block_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      pofHash: {
        constraints: {
          Pattern: new RegExp("^no_fraud|bf_\\w{38,50}$")
        },
        serializedName: "pof_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      prevHash: {
        serializedName: "prev_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      prevKeyHash: {
        constraints: {
          Pattern: new RegExp("^kh_\\w{38,50}$")
        },
        serializedName: "prev_key_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      signature: {
        constraints: {
          Pattern: new RegExp("^sg_\\w+$")
        },
        serializedName: "signature",
        required: true,
        type: {
          name: "String"
        }
      },
      stateHash: {
        constraints: {
          Pattern: new RegExp("^bs_\\w{38,50}$")
        },
        serializedName: "state_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      time: {
        serializedName: "time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      transactionsCount: {
        serializedName: "transactions_count",
        required: true,
        type: {
          name: "Number"
        }
      },
      txsHash: {
        constraints: {
          Pattern: new RegExp("^bx_\\w{38,50}$")
        },
        serializedName: "txs_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      version: {
        serializedName: "version",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsXhlqwrV3MicroBlocksHashTransactionsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsXhlqwrV3MicroBlocksHashTransactionsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Transaction"
            }
          }
        }
      }
    }
  }
};
export const Transaction = {
  type: {
    name: "Composite",
    className: "Transaction",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      encodedTx: {
        constraints: {
          Pattern: new RegExp("^tx_\\w+$")
        },
        serializedName: "encoded_tx",
        required: true,
        type: {
          name: "String"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      signatures: {
        serializedName: "signatures",
        required: true,
        type: {
          name: "Sequence",
          element: {
            constraints: {
              Pattern: new RegExp("^sg_\\w+$")
            },
            type: {
              name: "String"
            }
          }
        }
      },
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const Paths181Cs9V3NamesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths181Cs9V3NamesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Name"
            }
          }
        }
      }
    }
  }
};
export const Name = {
  type: {
    name: "Composite",
    className: "Name",
    modelProperties: {
      active: {
        serializedName: "active",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      activeFrom: {
        serializedName: "active_from",
        required: true,
        type: {
          name: "Number"
        }
      },
      approximateActivationTime: {
        serializedName: "approximate_activation_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      approximateExpireTime: {
        serializedName: "approximate_expire_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      auction: {
        serializedName: "auction",
        type: {
          name: "Composite",
          className: "Auction"
        }
      },
      auctionTimeout: {
        serializedName: "auction_timeout",
        required: true,
        type: {
          name: "Number"
        }
      },
      expireHeight: {
        serializedName: "expire_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^nm_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      name: {
        constraints: {
          Pattern: new RegExp("^\\w+\\.chain$")
        },
        serializedName: "name",
        required: true,
        type: {
          name: "String"
        }
      },
      nameFee: {
        serializedName: "name_fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      ownership: {
        serializedName: "ownership",
        type: {
          name: "Composite",
          className: "NameOwnership"
        }
      },
      pointers: {
        serializedName: "pointers",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      revoke: {
        serializedName: "revoke",
        type: {
          name: "Composite",
          className: "NameTx"
        }
      }
    }
  }
};
export const Auction = {
  type: {
    name: "Composite",
    className: "Auction",
    modelProperties: {
      activationTime: {
        serializedName: "activation_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      approximateExpireTime: {
        serializedName: "approximate_expire_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      auctionEnd: {
        serializedName: "auction_end",
        required: true,
        type: {
          name: "Number"
        }
      },
      lastBid: {
        serializedName: "last_bid",
        type: {
          name: "Composite",
          className: "AuctionLastBid"
        }
      },
      name: {
        constraints: {
          Pattern: new RegExp("^\\w+\\.chain$")
        },
        serializedName: "name",
        required: true,
        type: {
          name: "String"
        }
      },
      nameFee: {
        serializedName: "name_fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      }
    }
  }
};
export const AuctionLastBid = {
  type: {
    name: "Composite",
    className: "AuctionLastBid",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      encodedTx: {
        constraints: {
          Pattern: new RegExp("^tx_\\w+$")
        },
        serializedName: "encoded_tx",
        required: true,
        type: {
          name: "String"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      signatures: {
        serializedName: "signatures",
        required: true,
        type: {
          name: "Sequence",
          element: {
            constraints: {
              Pattern: new RegExp("^sg_\\w+$")
            },
            type: {
              name: "String"
            }
          }
        }
      },
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const NameOwnership = {
  type: {
    name: "Composite",
    className: "NameOwnership",
    modelProperties: {
      current: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "current",
        type: {
          name: "String"
        }
      },
      original: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "original",
        type: {
          name: "String"
        }
      }
    }
  }
};
export const NameTx = {
  type: {
    name: "Composite",
    className: "NameTx",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      signatures: {
        serializedName: "signatures",
        required: true,
        type: {
          name: "Sequence",
          element: {
            constraints: {
              Pattern: new RegExp("^sg_\\w+$")
            },
            type: {
              name: "String"
            }
          }
        }
      },
      tx: {
        serializedName: "tx",
        type: {
          name: "Composite",
          className: "NameTxTx"
        }
      }
    }
  }
};
export const NameTxTx = {
  type: {
    name: "Composite",
    className: "NameTxTx",
    modelProperties: {
      accountId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "account_id",
        required: true,
        type: {
          name: "String"
        }
      },
      fee: {
        serializedName: "fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      nameId: {
        constraints: {
          Pattern: new RegExp("^nm_\\w{38,50}$")
        },
        serializedName: "name_id",
        required: true,
        type: {
          name: "String"
        }
      },
      nonce: {
        serializedName: "nonce",
        required: true,
        type: {
          name: "Number"
        }
      },
      ttl: {
        serializedName: "ttl",
        type: {
          name: "Number"
        }
      },
      type: {
        serializedName: "type",
        required: true,
        type: {
          name: "String"
        }
      },
      version: {
        serializedName: "version",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1R08F8HV3NamesAuctionsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1R08F8HV3NamesAuctionsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Auction"
            }
          }
        }
      }
    }
  }
};
export const PathsCrb9BgV3NamesAuctionsIdClaimsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsCrb9BgV3NamesAuctionsIdClaimsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "NameClaim"
            }
          }
        }
      }
    }
  }
};
export const NameClaim = {
  type: {
    name: "Composite",
    className: "NameClaim",
    modelProperties: {
      activeFrom: {
        serializedName: "active_from",
        required: true,
        type: {
          name: "Number"
        }
      },
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      internalSource: {
        serializedName: "internal_source",
        type: {
          name: "Boolean"
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const PathsMyl4W2V3NamesIdClaimsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsMyl4W2V3NamesIdClaimsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "NameClaim"
            }
          }
        }
      }
    }
  }
};
export const Paths1Ukwk06V3NamesIdTransfersGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Ukwk06V3NamesIdTransfersGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "NameTransfer"
            }
          }
        }
      }
    }
  }
};
export const NameTransfer = {
  type: {
    name: "Composite",
    className: "NameTransfer",
    modelProperties: {
      activeFrom: {
        serializedName: "active_from",
        required: true,
        type: {
          name: "Number"
        }
      },
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      internalSource: {
        serializedName: "internal_source",
        type: {
          name: "Boolean"
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const PathsRcnvllV3NamesIdUpdatesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsRcnvllV3NamesIdUpdatesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "NameUpdate"
            }
          }
        }
      }
    }
  }
};
export const NameUpdate = {
  type: {
    name: "Composite",
    className: "NameUpdate",
    modelProperties: {
      activeFrom: {
        serializedName: "active_from",
        required: true,
        type: {
          name: "Number"
        }
      },
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      internalSource: {
        serializedName: "internal_source",
        type: {
          name: "Boolean"
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      }
    }
  }
};
export const PathsGcr9MrV3OraclesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsGcr9MrV3OraclesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Oracle"
            }
          }
        }
      }
    }
  }
};
export const Oracle = {
  type: {
    name: "Composite",
    className: "Oracle",
    modelProperties: {
      active: {
        serializedName: "active",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      activeFrom: {
        serializedName: "active_from",
        required: true,
        type: {
          name: "Number"
        }
      },
      approximateExpireTime: {
        serializedName: "approximate_expire_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      expireHeight: {
        serializedName: "expire_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      format: {
        serializedName: "format",
        type: {
          name: "Composite",
          className: "OracleFormat"
        }
      },
      oracle: {
        constraints: {
          Pattern: new RegExp("^ok_\\w{38,50}$")
        },
        serializedName: "oracle",
        required: true,
        type: {
          name: "String"
        }
      },
      queryFee: {
        serializedName: "query_fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      register: {
        serializedName: "register",
        type: {
          name: "Composite",
          className: "OracleTx"
        }
      },
      registerTime: {
        serializedName: "register_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      registerTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "register_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const OracleFormat = {
  type: {
    name: "Composite",
    className: "OracleFormat",
    modelProperties: {
      query: {
        serializedName: "query",
        required: true,
        type: {
          name: "String"
        }
      },
      response: {
        serializedName: "response",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const OracleTx = {
  type: {
    name: "Composite",
    className: "OracleTx",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      encodedTx: {
        constraints: {
          Pattern: new RegExp("^tx_\\w+$")
        },
        serializedName: "encoded_tx",
        required: true,
        type: {
          name: "String"
        }
      },
      hash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "hash",
        required: true,
        type: {
          name: "String"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      signatures: {
        serializedName: "signatures",
        required: true,
        type: {
          name: "Sequence",
          element: {
            constraints: {
              Pattern: new RegExp("^sg_\\w+$")
            },
            type: {
              name: "String"
            }
          }
        }
      },
      tx: {
        serializedName: "tx",
        type: {
          name: "Composite",
          className: "OracleTxTx"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const OracleTxTx = {
  type: {
    name: "Composite",
    className: "OracleTxTx",
    modelProperties: {
      abiVersion: {
        serializedName: "abi_version",
        required: true,
        type: {
          name: "Number"
        }
      },
      accountId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "account_id",
        required: true,
        type: {
          name: "String"
        }
      },
      fee: {
        serializedName: "fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      nonce: {
        serializedName: "nonce",
        required: true,
        type: {
          name: "Number"
        }
      },
      oracleId: {
        constraints: {
          Pattern: new RegExp("^ok_\\w{38,50}$")
        },
        serializedName: "oracle_id",
        required: true,
        type: {
          name: "String"
        }
      },
      oracleTtl: {
        serializedName: "oracle_ttl",
        type: {
          name: "Composite",
          className: "OracleTxOracleTtl"
        }
      },
      queryFee: {
        serializedName: "query_fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      queryFormat: {
        serializedName: "query_format",
        required: true,
        type: {
          name: "String"
        }
      },
      responseFormat: {
        serializedName: "response_format",
        required: true,
        type: {
          name: "String"
        }
      },
      ttl: {
        serializedName: "ttl",
        type: {
          name: "Number"
        }
      },
      type: {
        serializedName: "type",
        required: true,
        type: {
          name: "String"
        }
      },
      version: {
        serializedName: "version",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const OracleTxOracleTtl = {
  type: {
    name: "Composite",
    className: "OracleTxOracleTtl",
    modelProperties: {
      type: {
        serializedName: "type",
        required: true,
        type: {
          name: "String"
        }
      },
      value: {
        serializedName: "value",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsZ4L2QlV3OraclesIdExtendsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsZ4L2QlV3OraclesIdExtendsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "OracleExtend"
            }
          }
        }
      }
    }
  }
};
export const OracleExtend = {
  type: {
    name: "Composite",
    className: "OracleExtend",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      tx: {
        serializedName: "tx",
        type: {
          name: "Composite",
          className: "OracleExtendTx"
        }
      }
    }
  }
};
export const OracleExtendTx = {
  type: {
    name: "Composite",
    className: "OracleExtendTx",
    modelProperties: {
      fee: {
        serializedName: "fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      nonce: {
        serializedName: "nonce",
        required: true,
        type: {
          name: "Number"
        }
      },
      oracleId: {
        constraints: {
          Pattern: new RegExp("^ok_\\w{38,50}$")
        },
        serializedName: "oracle_id",
        required: true,
        type: {
          name: "String"
        }
      },
      oracleTtl: {
        serializedName: "oracle_ttl",
        type: {
          name: "Composite",
          className: "OracleExtendTxOracleTtl"
        }
      },
      ttl: {
        serializedName: "ttl",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const OracleExtendTxOracleTtl = {
  type: {
    name: "Composite",
    className: "OracleExtendTxOracleTtl",
    modelProperties: {
      type: {
        serializedName: "type",
        required: true,
        type: {
          name: "String"
        }
      },
      value: {
        serializedName: "value",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1Uni7AtV3OraclesIdQueriesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Uni7AtV3OraclesIdQueriesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "OracleQuery"
            }
          }
        }
      }
    }
  }
};
export const OracleQuery = {
  type: {
    name: "Composite",
    className: "OracleQuery",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockTime: {
        serializedName: "block_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      fee: {
        serializedName: "fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      nonce: {
        serializedName: "nonce",
        required: true,
        type: {
          name: "Number"
        }
      },
      oracleId: {
        constraints: {
          Pattern: new RegExp("^ok_\\w{38,50}$")
        },
        serializedName: "oracle_id",
        required: true,
        type: {
          name: "String"
        }
      },
      query: {
        serializedName: "query",
        required: true,
        type: {
          name: "String"
        }
      },
      queryFee: {
        serializedName: "query_fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      queryId: {
        constraints: {
          Pattern: new RegExp("^oq_\\w{38,50}$")
        },
        serializedName: "query_id",
        required: true,
        type: {
          name: "String"
        }
      },
      queryTtl: {
        serializedName: "query_ttl",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      response: {
        serializedName: "response",
        type: {
          name: "Composite",
          className: "OracleResponse"
        }
      },
      responseTtl: {
        serializedName: "response_ttl",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      senderId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "sender_id",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      ttl: {
        serializedName: "ttl",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const OracleResponse = {
  type: {
    name: "Composite",
    className: "OracleResponse",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      blockTime: {
        serializedName: "block_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      fee: {
        serializedName: "fee",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      nonce: {
        serializedName: "nonce",
        required: true,
        type: {
          name: "Number"
        }
      },
      oracleId: {
        constraints: {
          Pattern: new RegExp("^ok_\\w{38,50}$")
        },
        serializedName: "oracle_id",
        required: true,
        type: {
          name: "String"
        }
      },
      query: {
        serializedName: "query",
        type: {
          name: "Composite",
          className: "OracleResponse"
        }
      },
      queryId: {
        constraints: {
          Pattern: new RegExp("^oq_\\w{38,50}$")
        },
        serializedName: "query_id",
        required: true,
        type: {
          name: "String"
        }
      },
      response: {
        serializedName: "response",
        required: true,
        type: {
          name: "String"
        }
      },
      responseTtl: {
        serializedName: "response_ttl",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      sourceTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "source_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      sourceTxType: {
        serializedName: "source_tx_type",
        required: true,
        type: {
          name: "String"
        }
      },
      ttl: {
        serializedName: "ttl",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths1Tcj5A9V3OraclesIdResponsesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Tcj5A9V3OraclesIdResponsesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "OracleResponse"
            }
          }
        }
      }
    }
  }
};
export const Stats = {
  type: {
    name: "Composite",
    className: "Stats",
    modelProperties: {
      feesTrend: {
        serializedName: "fees_trend",
        type: {
          name: "Number"
        }
      },
      last24HsAverageTransactionFees: {
        serializedName: "last_24hs_average_transaction_fees",
        type: {
          name: "Number"
        }
      },
      last24HsTransactions: {
        serializedName: "last_24hs_transactions",
        type: {
          name: "Number"
        }
      },
      maxTransactionsPerSecond: {
        serializedName: "max_transactions_per_second",
        type: {
          name: "Number"
        }
      },
      maxTransactionsPerSecondBlockHash: {
        constraints: {
          Pattern: new RegExp("^kh_\\w{38,50}$")
        },
        serializedName: "max_transactions_per_second_block_hash",
        type: {
          name: "String"
        }
      },
      minersCount: {
        serializedName: "miners_count",
        type: {
          name: "Number"
        }
      },
      transactionsTrend: {
        serializedName: "transactions_trend",
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsV2Gh3TV3StatisticsBlocksGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsV2Gh3TV3StatisticsBlocksGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Stat"
            }
          }
        }
      }
    }
  }
};
export const Stat = {
  type: {
    name: "Composite",
    className: "Stat",
    modelProperties: {
      count: {
        serializedName: "count",
        required: true,
        type: {
          name: "Number"
        }
      },
      endDate: {
        serializedName: "end_date",
        required: true,
        type: {
          name: "String"
        }
      },
      startDate: {
        serializedName: "start_date",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const PathsYpljbvV3DeltastatsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsYpljbvV3DeltastatsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "DeltaStat"
            }
          }
        }
      }
    }
  }
};
export const DeltaStat = {
  type: {
    name: "Composite",
    className: "DeltaStat",
    modelProperties: {
      auctionsStarted: {
        serializedName: "auctions_started",
        required: true,
        type: {
          name: "Number"
        }
      },
      blockReward: {
        serializedName: "block_reward",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      burnedInAuctions: {
        serializedName: "burned_in_auctions",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      channelsClosed: {
        serializedName: "channels_closed",
        required: true,
        type: {
          name: "Number"
        }
      },
      channelsOpened: {
        serializedName: "channels_opened",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractsCreated: {
        serializedName: "contracts_created",
        required: true,
        type: {
          name: "Number"
        }
      },
      devReward: {
        serializedName: "dev_reward",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      lastTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "last_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      lockedInAuctions: {
        serializedName: "locked_in_auctions",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      lockedInChannels: {
        serializedName: "locked_in_channels",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      namesActivated: {
        serializedName: "names_activated",
        required: true,
        type: {
          name: "Number"
        }
      },
      namesExpired: {
        serializedName: "names_expired",
        required: true,
        type: {
          name: "Number"
        }
      },
      namesRevoked: {
        serializedName: "names_revoked",
        required: true,
        type: {
          name: "Number"
        }
      },
      oraclesExpired: {
        serializedName: "oracles_expired",
        required: true,
        type: {
          name: "Number"
        }
      },
      oraclesRegistered: {
        serializedName: "oracles_registered",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const Paths10OmbqhV3MinerstatsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths10OmbqhV3MinerstatsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Miner"
            }
          }
        }
      }
    }
  }
};
export const Miner = {
  type: {
    name: "Composite",
    className: "Miner",
    modelProperties: {
      miner: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "miner",
        required: true,
        type: {
          name: "String"
        }
      },
      totalReward: {
        serializedName: "total_reward",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      }
    }
  }
};
export const PathsAc89GqV3StatisticsNamesGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsAc89GqV3StatisticsNamesGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Stat"
            }
          }
        }
      }
    }
  }
};
export const Paths1Vms155V3TotalstatsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Vms155V3TotalstatsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "TotalStat"
            }
          }
        }
      }
    }
  }
};
export const TotalStat = {
  type: {
    name: "Composite",
    className: "TotalStat",
    modelProperties: {
      activeAuctions: {
        serializedName: "active_auctions",
        required: true,
        type: {
          name: "Number"
        }
      },
      activeNames: {
        serializedName: "active_names",
        required: true,
        type: {
          name: "Number"
        }
      },
      activeOracles: {
        serializedName: "active_oracles",
        required: true,
        type: {
          name: "Number"
        }
      },
      burnedInAuctions: {
        serializedName: "burned_in_auctions",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      contracts: {
        serializedName: "contracts",
        required: true,
        type: {
          name: "Number"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      inactiveNames: {
        serializedName: "inactive_names",
        required: true,
        type: {
          name: "Number"
        }
      },
      inactiveOracles: {
        serializedName: "inactive_oracles",
        required: true,
        type: {
          name: "Number"
        }
      },
      lastTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "last_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      lockedInAuctions: {
        serializedName: "locked_in_auctions",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      lockedInChannels: {
        serializedName: "locked_in_channels",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      openChannels: {
        serializedName: "open_channels",
        required: true,
        type: {
          name: "Number"
        }
      },
      sumBlockReward: {
        serializedName: "sum_block_reward",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      sumDevReward: {
        serializedName: "sum_dev_reward",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      totalTokenSupply: {
        serializedName: "total_token_supply",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      }
    }
  }
};
export const Paths1Syc8AnV3StatisticsTransactionsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "Paths1Syc8AnV3StatisticsTransactionsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Stat"
            }
          }
        }
      }
    }
  }
};
export const Status = {
  type: {
    name: "Composite",
    className: "Status",
    modelProperties: {
      mdwAsyncTasks: {
        serializedName: "mdw_async_tasks",
        type: {
          name: "Composite",
          className: "StatusMdwAsyncTasks"
        }
      },
      mdwGensPerMinute: {
        serializedName: "mdw_gens_per_minute",
        required: true,
        type: {
          name: "Number"
        }
      },
      mdwHeight: {
        serializedName: "mdw_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      mdwLastMigration: {
        serializedName: "mdw_last_migration",
        required: true,
        type: {
          name: "Number"
        }
      },
      mdwRevision: {
        serializedName: "mdw_revision",
        required: true,
        type: {
          name: "String"
        }
      },
      mdwSynced: {
        serializedName: "mdw_synced",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      mdwSyncing: {
        serializedName: "mdw_syncing",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      mdwTxIndex: {
        serializedName: "mdw_tx_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      mdwVersion: {
        serializedName: "mdw_version",
        required: true,
        type: {
          name: "String"
        }
      },
      nodeHeight: {
        serializedName: "node_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      nodeProgress: {
        serializedName: "node_progress",
        required: true,
        type: {
          name: "Number"
        }
      },
      nodeRevision: {
        serializedName: "node_revision",
        required: true,
        type: {
          name: "String"
        }
      },
      nodeSyncing: {
        serializedName: "node_syncing",
        required: true,
        type: {
          name: "Boolean"
        }
      },
      nodeVersion: {
        serializedName: "node_version",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const StatusMdwAsyncTasks = {
  type: {
    name: "Composite",
    className: "StatusMdwAsyncTasks",
    modelProperties: {
      longTasks: {
        serializedName: "long_tasks",
        required: true,
        type: {
          name: "Number"
        }
      },
      producerBuffer: {
        serializedName: "producer_buffer",
        required: true,
        type: {
          name: "Number"
        }
      },
      totalPending: {
        serializedName: "total_pending",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const PathsHa9C78V3TransactionsGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsHa9C78V3TransactionsGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Transaction"
            }
          }
        }
      }
    }
  }
};
export const PathsVdg67V3TransfersGetResponses200ContentApplicationJsonSchemaAllof0 = {
  type: {
    name: "Composite",
    className: "PathsVdg67V3TransfersGetResponses200ContentApplicationJsonSchemaAllof0",
    modelProperties: {
      data: {
        serializedName: "data",
        required: true,
        type: {
          name: "Sequence",
          element: {
            type: {
              name: "Composite",
              className: "Transfer"
            }
          }
        }
      }
    }
  }
};
export const Transfer = {
  type: {
    name: "Composite",
    className: "Transfer",
    modelProperties: {
      accountId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "account_id",
        required: true,
        type: {
          name: "String"
        }
      },
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      kind: {
        serializedName: "kind",
        required: true,
        type: {
          name: "String"
        }
      },
      refBlockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "ref_block_hash",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      refTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "ref_tx_hash",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      },
      refTxType: {
        serializedName: "ref_tx_type",
        required: true,
        nullable: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Aex9TransferEvent = {
  type: {
    name: "Composite",
    className: "Aex9TransferEvent",
    modelProperties: {
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      recipientId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "recipient_id",
        required: true,
        type: {
          name: "String"
        }
      },
      senderId: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "sender_id",
        required: true,
        type: {
          name: "String"
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Aex9TransferResponse = {
  type: {
    name: "Composite",
    className: "Aex9TransferResponse",
    modelProperties: {
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      blockHeight: {
        serializedName: "block_height",
        required: true,
        type: {
          name: "Number"
        }
      },
      callTxi: {
        serializedName: "call_txi",
        required: true,
        type: {
          name: "Number"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      logIdx: {
        serializedName: "log_idx",
        required: true,
        type: {
          name: "Number"
        }
      },
      microTime: {
        serializedName: "micro_time",
        required: true,
        type: {
          name: "UnixTime"
        }
      },
      recipient: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "recipient",
        required: true,
        type: {
          name: "String"
        }
      },
      sender: {
        constraints: {
          Pattern: new RegExp("^ak_\\w{38,50}$")
        },
        serializedName: "sender",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const InternalContractCallEvent = {
  type: {
    name: "Composite",
    className: "InternalContractCallEvent",
    modelProperties: {
      blockHash: {
        constraints: {
          Pattern: new RegExp("^mh_\\w{38,50}$")
        },
        serializedName: "block_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      callTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "call_tx_hash",
        type: {
          name: "String"
        }
      },
      contractId: {
        constraints: {
          Pattern: new RegExp("^ct_\\w{38,50}$")
        },
        serializedName: "contract_id",
        required: true,
        type: {
          name: "String"
        }
      },
      contractTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "contract_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      },
      function: {
        serializedName: "function",
        required: true,
        type: {
          name: "String"
        }
      },
      height: {
        serializedName: "height",
        required: true,
        type: {
          name: "Number"
        }
      },
      internalTx: {
        serializedName: "internal_tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      microIndex: {
        serializedName: "micro_index",
        required: true,
        type: {
          name: "Number"
        }
      }
    }
  }
};
export const InternalTransferEvent = {
  type: {
    name: "Composite",
    className: "InternalTransferEvent",
    modelProperties: {
      amount: {
        serializedName: "amount",
        required: true,
        type: {
          // @ts-expect-error we are extending autorest with BigInt support
          name: "BigInt"
        }
      },
      kind: {
        serializedName: "kind",
        required: true,
        type: {
          name: "String"
        }
      },
      refTxHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "ref_tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const NameClaimEvent = {
  type: {
    name: "Composite",
    className: "NameClaimEvent",
    modelProperties: {
      tx: {
        serializedName: "tx",
        required: true,
        type: {
          name: "Dictionary",
          value: {
            type: {
              name: "any"
            }
          }
        }
      },
      txHash: {
        constraints: {
          Pattern: new RegExp("^th_\\w{38,50}$")
        },
        serializedName: "tx_hash",
        required: true,
        type: {
          name: "String"
        }
      }
    }
  }
};
export const Paths3Hsv3GV3AccountsAccountidAex141TokensGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths3Hsv3GV3AccountsAccountidAex141TokensGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Q9E32FV3AccountsAccountidAex141TokensGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsKm52GqV3AccountsAccountidAex9BalancesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsKm52GqV3AccountsAccountidAex9BalancesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PaginatedResponse.type.modelProperties,
      ...PathsZ92TkfV3AccountsAccountidAex9BalancesGetResponses200ContentApplicationJsonSchemaAllof1.type.modelProperties
    }
  }
};
export const Paths1Ceolv9V3AccountsAccountIdDexSwapsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Ceolv9V3AccountsAccountIdDexSwapsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Ymnu9GV3AccountsAccountIdDexSwapsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Opead5V3AccountsIdActivitiesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Opead5V3AccountsIdActivitiesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Ent1R1V3AccountsIdActivitiesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Gc4HwtV3AccountsIdNamesPointeesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Gc4HwtV3AccountsIdNamesPointeesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsC0UvfgV3AccountsIdNamesPointeesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1XwlyjtV3Aex141GetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1XwlyjtV3Aex141GetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths8I0YgwV3Aex141GetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths181AjwxV3Aex141ContractidTemplatesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths181AjwxV3Aex141ContractidTemplatesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsWkpcwaV3Aex141ContractidTemplatesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths15Mi2TaV3Aex141ContractidTemplatesTemplateidTokensGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths15Mi2TaV3Aex141ContractidTemplatesTemplateidTokensGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsRay4X0V3Aex141ContractidTemplatesTemplateidTokensGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsWl652MV3Aex141ContractidTokensGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsWl652MV3Aex141ContractidTokensGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1TkisghV3Aex141ContractidTokensGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1O7Q6IhV3Aex141ContractidTransfersGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1O7Q6IhV3Aex141ContractidTransfersGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1A8Ah39V3Aex141ContractidTransfersGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Vr3Y2EV3Aex9GetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Vr3Y2EV3Aex9GetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Uqqby0V3Aex9GetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1N61UurV3Aex9ContractidBalancesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1N61UurV3Aex9ContractidBalancesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsEeiffwV3Aex9ContractidBalancesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Uybd4PV3Aex9ContractidBalancesAccountidHistoryGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Uybd4PV3Aex9ContractidBalancesAccountidHistoryGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PaginatedResponse.type.modelProperties,
      ...Paths108B3VtV3Aex9ContractidBalancesAccountidHistoryGetResponses200ContentApplicationJsonSchemaAllof1.type.modelProperties
    }
  }
};
export const Paths3EzhapV3ChannelsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths3EzhapV3ChannelsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsQmewnaV3ChannelsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Txblx8V3ContractsCallsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Txblx8V3ContractsCallsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths18L84JcV3ContractsCallsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths7A1M6RV3ContractsLogsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths7A1M6RV3ContractsLogsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths10Kk1UxV3ContractsLogsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Di8FnjV3DexSwapsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Di8FnjV3DexSwapsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths9Yfxl2V3DexSwapsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsKwxlzlV3DexContractIdSwapsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsKwxlzlV3DexContractIdSwapsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths6Vze8ZV3DexContractIdSwapsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths277OngV3KeyBlocksGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths277OngV3KeyBlocksGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsEue6HzV3KeyBlocksGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1159W94V3KeyBlocksHashOrKbiMicroBlocksGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1159W94V3KeyBlocksHashOrKbiMicroBlocksGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsNn60D7V3KeyBlocksHashOrKbiMicroBlocksGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths15Bkk50V3MicroBlocksHashTransactionsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths15Bkk50V3MicroBlocksHashTransactionsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsXhlqwrV3MicroBlocksHashTransactionsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths12S1Nd4V3NamesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths12S1Nd4V3NamesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths181Cs9V3NamesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsKjq4D4V3NamesAuctionsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsKjq4D4V3NamesAuctionsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1R08F8HV3NamesAuctionsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1F98AqgV3NamesIdClaimsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1F98AqgV3NamesIdClaimsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsMyl4W2V3NamesIdClaimsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Raw8PV3NamesIdTransfersGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Raw8PV3NamesIdTransfersGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Ukwk06V3NamesIdTransfersGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Ec8CltV3NamesIdUpdatesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Ec8CltV3NamesIdUpdatesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsRcnvllV3NamesIdUpdatesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1E14NekV3OraclesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1E14NekV3OraclesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsGcr9MrV3OraclesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1L5C64RV3OraclesIdExtendsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1L5C64RV3OraclesIdExtendsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsZ4L2QlV3OraclesIdExtendsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths8722JhV3OraclesIdQueriesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths8722JhV3OraclesIdQueriesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Uni7AtV3OraclesIdQueriesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsVron83V3OraclesIdResponsesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsVron83V3OraclesIdResponsesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Tcj5A9V3OraclesIdResponsesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsJd68YV3StatisticsBlocksGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsJd68YV3StatisticsBlocksGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsV2Gh3TV3StatisticsBlocksGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsQtodvgV3DeltastatsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsQtodvgV3DeltastatsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsYpljbvV3DeltastatsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsZ2B507V3MinerstatsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsZ2B507V3MinerstatsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths10OmbqhV3MinerstatsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsD2HmjxV3StatisticsNamesGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsD2HmjxV3StatisticsNamesGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsAc89GqV3StatisticsNamesGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsFrvc1LV3TotalstatsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsFrvc1LV3TotalstatsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Vms155V3TotalstatsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths150Ou6YV3StatisticsTransactionsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths150Ou6YV3StatisticsTransactionsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...Paths1Syc8AnV3StatisticsTransactionsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const Paths1Pymq07V3TransactionsGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "Paths1Pymq07V3TransactionsGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsHa9C78V3TransactionsGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
export const PathsA7P0KiV3TransfersGetResponses200ContentApplicationJsonSchema = {
  type: {
    name: "Composite",
    className: "PathsA7P0KiV3TransfersGetResponses200ContentApplicationJsonSchema",
    modelProperties: {
      ...PathsVdg67V3TransfersGetResponses200ContentApplicationJsonSchemaAllof0.type.modelProperties,
      ...PaginatedResponse.type.modelProperties
    }
  }
};
